pkg/feeds/epochs/epoch.go

```diff
@@ -0,0 +1,84 @@
+package epochs
+
+import (
+	"encoding/binary"
+
+	"github.com/ethersphere/bee/pkg/crypto"
+)
+
+const (
+	maxLevel = 32
+)
+
+// epoch is referencing a slot in the epoch grid
+type epoch struct {
+	start uint64
+	level uint8
+}
+
+// MarshalBinary implements the BinaryMarshaler interface
+func (e *epoch) MarshalBinary() ([]byte, error) {
+	epochBytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(epochBytes, e.start)
+	return crypto.LegacyKeccak256(append(epochBytes, e.level))
+}
+
+// lca calculates the lowest common ancestor epoch given two unix times
+func lca(at, after int64) *epoch {
```

Comment:
none of the methods in this package are unit tested, so it is difficult for the reader of the code to understand how each and every one of them work separately. they are only tested using generalised e2e tests that are binary - they either pass or they don't. if they were to fail one day, maintainers would either need to onboard to a lot of untested code, which is sub optimal. i would expect at least a bit of unit testing, even if only for sake of understanding how each method plays separately.

Reply:
^ technical debt TODO